unit-tests/dds/adapter/test-depth.py

```diff
@@ -0,0 +1,86 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#test:donotrun:!dds
+#test:device D400*
+
+from rspy import log, repo, test
+import time
+
+
+with test.closure( 'Run rs-dds-adapter', on_fail=test.ABORT ):
+    adapter_path = repo.find_built_exe( 'tools/dds/dds-adapter', 'rs-dds-adapter' )
+    if test.check( adapter_path ):
+        import subprocess, signal
+        cmd = [adapter_path, '--domain-id', '123']
+        if log.is_debug_on():
+            cmd.append( '--debug' )
+        adapter_process = subprocess.Popen( cmd,
+            stdout=None,
+            stderr=subprocess.STDOUT,
+            universal_newlines=True )  # don't fail on errors
+
+from rspy import librs as rs
+if log.is_debug_on():
+    rs.log_to_console( rs.log_severity.debug )
+
+with test.closure( 'Initialize librealsense context', on_fail=test.ABORT ):
+    context = rs.context( { 'dds': { 'enabled': True, 'domain': 123, 'participant': 'client' }} )
+
+with test.closure( 'Wait for a device', on_fail=test.ABORT ):
+    # Note: takes time for a device to enumerate, and more to get it discovered
+    dev = rs.wait_for_devices( context, rs.only_sw_devices, n=1., timeout=8 )
+
+with test.closure( 'Get sensor', on_fail=test.ABORT ):
+    sensor = dev.first_depth_sensor()
+    test.check( sensor )
+
+with test.closure( 'Find profile', on_fail=test.ABORT ):
+    for p in sensor.profiles:
+        log.d( p )
+    profile = next( p for p in sensor.profiles
+                    if p.fps() == 30
+                    and p.stream_type() == rs.stream.depth )
+    test.check( profile )
+
+n_frames = 0
+start_time = None
+def frame_callback( frame ):
+    global n_frames, start_time
+    if n_frames == 0:
+        start_time = time.perf_counter()
+    n_frames += 1
+
+with test.closure( f'Stream {profile}', on_fail=test.ABORT ):
+    sensor.open( [profile] )
+    sensor.start( frame_callback )
+
+with test.closure( 'Let it stream' ):
+    time.sleep( 3 )
+    end_time = time.perf_counter()
+    if test.check( n_frames > 0 ):
+        test.info( 'start_time', start_time )
+        test.info( 'end_time', end_time )
+        test.info( 'n_frames', n_frames )
+        test.check_approx_abs( n_frames / (end_time-start_time), 29.5, 1 )
```

Comment:
Validation FPS KPI is 5%, for 30Hz can deviate by up to 1.5Hz.
Already seen a failure with ~28.95Hz, so better update this

Reply:
It was 30 +/- 1, which is why it failed on 28.95 which is why I changed.
It's now 28.5-30.5 (I've seen it over 30, too). I think it should be good, but if not, what exactly would you like here? 28.5-31.5?

Reply:
Validation tests for 28.5-31.5, but FPS over 30 is less likely.
Leave it 28.5-30.5, if we will see issues with too high FPS we can adjust the test.